src/webview/MessageList.js

```diff
@@ -365,7 +365,7 @@ export default connect<SelectorProps, _, _>((state, props: OuterProps) => {
       ? { older: false, newer: false }
       : getFetchingForNarrow(state, props.narrow),
     htmlPieceDescriptorsForShownMessages: props.htmlPieceDescriptorsForShownMessages
-      ? props.htmlPieceDescriptorsForShownMessages
+      ? getHtmlPieceDescriptorsForMessages(props.messages, props.narrow)
       : getHtmlPieceDescriptorsForMessages(props.messages, props.narrow),
```

Comment:
SearchMessagesCard [nfc]: Don't pass htmlPieceDescriptorsForShownMessages.
    
    Instead, have the conditional branch in `MessageList` that was
    exercised by passing this prop compute the value it needs for
    itself.

I'm a bit confused by this commit. We stop passing the prop in this caller; and then we also change the behavior in the case where the prop *is* passed?

(But at this point there is no case where it's passed, right?)

I think this and the next two commits can perhaps be clearer as just two commits:
* First, stop passing the prop in the search case (the only case where we were passing it.) This is NFC because the falsy branch of this conditional computes the same thing we had been passing.
* Then, eliminate the prop from the interface, and always just compute it here. This is NFC because at this point we're never passing the prop anyway.